### DISCUSSION TITLE
2021/a-powershell-script-for-activating-an-eligible-role-assignment-in-azure-ad/

### DISCUSSION THREAD

**Discussion by giscus** [mirontoli/chuvash-eu-comments#3] [General]
# 2021/a-powershell-script-for-activating-an-eligible-role-assignment-in-azure-ad/

Recently my role assignments in Azure AD were switched from permanent to eligible ones. This is part of PIM - Privileged Identity Management, you can read more about it on MS Docs:
Start using Privileged Identity Management To activate your eligible assignment you can use Azure Portal, Graph API, and PowerShell. The activation in the portal and Graph API is described on MS Docs:
Activate my Azure AD roles in PIM My roles within Privileged Identity Management in Azure Portal

https://chuvash.eu/2021/a-powershell-script-for-activating-an-eligible-role-assignment-in-azure-ad/

**Comment by NeelDarji1989** [comment]
Hello

This is such a great work!

Let say, I want that when end user calls this script, it should ask to choose which Eligible Role End user want to activate, then according to Role Settings, it should ask to enter Justification or Ticket Information and then script should activate the role.

Can you please help to prepare this?

I need for both - Entra as well as RBAC Roles

**Reply by mirontoli** [reply]
Thank you for your feedback! This is community effort, you are welcome to work further on this script to fit your needs and share it too. Sharing is caring.
